### PR TITLE
Reduce duplication of JSON guns [RDY]

### PR DESCRIPTION
**Pull request by mugling**
Depends upon functionality provided by #15933 and is intended to be merged concurrently with it.

Uses `copy-from` to reduce duplication where one gun is a derivative of another. In doing so corrects a number of inconsistencies.

Adds spawns for the `L523` conversion kits and differentiates each variant (0bc449e)

Finally aec2899 implements the `pistol_revolver` class so that all pistols now have a consistent set of `flags` and are `reload 75`

### CODE REVIEW

**Comment by Coolthulhu** [comment]
Not sure if it's this one, but it's most likely this one: I got conflict with Generic Guns regarding lw12mag and lw21mag.

**Comment by mugling** [comment]
It's likely to be this one so I'll take a look

**Comment by mugling** [comment]
658923f provides fix

File: data/json/items/gun/223.json [hunk 1]
```diff
@@ -151,138 +149,82 @@
...
-    "bashing": 12,
-    "to_hit": -1,
     "material": ["steel", "plastic"],
     "symbol": "(",
     "color": "dark_gray",
     "skill": "rifle",
     "ammo": "223",
-    "ranged_damage": -2,
-    "range": -12,
-    "dispersion": 150,
```
Comment (Coolthulhu) [inline data/json/items/gun/223.json]: The description states it is rather unwieldy without conversions, but it has really good stats - perfect accuracy, huge range, good damage, great recoil management etc.

**Comment by Coolthulhu** [comment]
The weird conversion rifle got buffed hard, it should be nerfed in some future update or get the description changed. It is currently really good - accuracy of a dedicated sniper rifle, but with bursts.

File: data/json/items/gun/223.json [hunk 2]
```diff
@@ -151,138 +149,82 @@
...
     "sight_dispersion": 60,
     "aim_speed" : 6,
-    "recoil": 60,
+    "recoil": -150,
     "durability": 3,
```
Comment (mugling) [inline data/json/items/gun/223.json]: It also has very low `durability` in-keeping with the description. Once we rework jams/misfires that's a harsh penalty whilst still making the gun usable which is perhaps more interesting than just making it a bad item.

**Comment by mugling** [comment]
> It is currently really good - accuracy of a dedicated sniper rifle, but with bursts.

Yes, should lose the bursts. I'll amend that in the next PR

**Pull request merged**